spark/src/main/scala/geotrellis/spark/join/VectorJoin.scala

```diff
@@ -0,0 +1,49 @@
+package geotrellis.spark.join
+
+import geotrellis.vector._
+import geotrellis.spark._
+import geotrellis.spark.tiling._
+
+import org.apache.spark._
+import org.apache.spark.rdd._
+import org.apache.spark.SparkContext._
+
+import scala.reflect._
+
+object VectorJoin {
+  /** Map each item to all the layout cells that it intersects with */
+  def mapToKeys[T <% Geometry](rdd: RDD[T], layout: LayoutDefinition): RDD[(SpatialKey, T)] = {
+    val mt = layout.mapTransform
+    rdd.flatMap{ thing =>
+      for { (col, row) <- mt((thing: Geometry).envelope).coords }
```

Comment:
Shouldn't this use the same technique that @jamesmcclain used to determine what keys overlapped a polygon? I can see many cases where this would produce a lot of keys that don't actually intersect oddly shaped polygons or multipolygons

https://github.com/geotrellis/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/LayerFilter.scala#L157